### ISSUE TITLE
Mist can't open. Ethereum node connection error....

### ISSUE THREAD

**Issue by jiiiiiiiin** [ethereum/mist#1042]
#### System information
``````
Version: `0.8.1`
OS & Version: Ubuntu 16.X (VMware)
Node type: `eth/geth(default)`
``````
![error](https://cloud.githubusercontent.com/assets/20676663/17166063/9ffc9898-540f-11e6-82ea-fff8fb3671e2.png)

```javascript
[2016-07-26 23:14:27.411] [INFO] main - Running in production mode: false
[2016-07-26 23:14:27.516] [INFO] main - Starting in Mist mode

(electron:2965): GLib-GIO-CRITICAL **: g_dbus_connection_register_object: assertion 'G_IS_DBUS_CONNECTION (connection)' failed

(electron:2965): GLib-GIO-CRITICAL **: g_dbus_connection_register_object: assertion 'G_IS_DBUS_CONNECTION (connection)' failed

(electron:2965): GLib-GIO-CRITICAL **: g_dbus_connection_get_unique_name: assertion 'G_IS_DBUS_CONNECTION (connection)' failed

(electron:2965): GConf-WARNING **: Client failed to connect to the D-BUS daemon:
Did not receive a reply. Possible causes include: the remote application did not send a reply, the message bus security policy blocked the reply, the reply timeout expired, or the network connection was broken.
[2016-07-26 23:14:27.768] [INFO] Db - Loading db: /root/.config/Mist/mist.lokidb
[2016-07-26 23:14:27.824] [INFO] Windows - Creating commonly-used windows
[2016-07-26 23:14:27.826] [INFO] Windows - Create secondary window: loading, owner: notset
[2016-07-26 23:14:27.871] [INFO] updateChecker - Check for update...
[2016-07-26 23:14:30.592] [INFO] Windows - Create primary window: main, owner: notset
[2016-07-26 23:14:30.637] [INFO] Windows - Create primary window: splash, owner: notset
[2016-07-26 23:14:31.926] [INFO] ipcCommunicator - Backend language set to:  en-US
[2016-07-26 23:14:33.700] [INFO] (ui: splash) - Web3 already initialized, re-using provider.
[2016-07-26 23:14:34.155] [INFO] Sockets/node-ipc - Connect to {"path":"/root/.ethereum/geth.ipc"}
[2016-07-26 23:14:34.263] [WARN] Sockets/node-ipc - Connection failed, retrying after 1000ms...
[2016-07-26 23:14:34.675] [INFO] updateChecker - App is up-to-date.
[2016-07-26 23:14:35.267] [WARN] Sockets/node-ipc - Connection failed, retrying after 1000ms...
[2016-07-26 23:14:36.269] [WARN] Sockets/node-ipc - Connection failed, retrying after 1000ms...
[2016-07-26 23:14:37.208] [ERROR] Sockets/node-ipc - Connection failed (3000ms elapsed)
[2016-07-26 23:14:37.215] [WARN] EthereumNode - Failed to connect to node. Maybe it's not running so let's start our own...
[2016-07-26 23:14:37.216] [INFO] EthereumNode - Node type: geth
[2016-07-26 23:14:37.217] [INFO] EthereumNode - Network: main
[2016-07-26 23:14:37.219] [INFO] EthereumNode - Start node: geth main
[2016-07-26 23:14:37.339] [INFO] EthereumNode - Node startup error
[2016-07-26 23:14:37.339] [ERROR] EthereumNode - Failed to start node { Error: spawn /home/islab/mist/nodes/geth/linux-x64/geth ENOENT
    at exports._errnoException (util.js:949:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:182:32)
    at onErrorNT (internal/child_process.js:348:16)
    at _combinedTickCallback (internal/process/next_tick.js:74:11)
    at process._tickCallback (internal/process/next_tick.js:98:9)
  code: 'ENOENT',
  errno: 'ENOENT',
  syscall: 'spawn /home/islab/mist/nodes/geth/linux-x64/geth',
  path: '/home/islab/mist/nodes/geth/linux-x64/geth',
  spawnargs: [ '--fast', '--cache', '512', '--support-dao-fork' ] }

(electron:2965): GLib-GIO-CRITICAL **: g_dbus_connection_register_object: assertion 'G_IS_DBUS_CONNECTION (connection)' failed

(electron:2965): GLib-GIO-CRITICAL **: g_dbus_connection_register_object: assertion 'G_IS_DBUS_CONNECTION (connection)' failed

(electron:2965): GLib-GIO-CRITICAL **: g_dbus_connection_get_unique_name: assertion 'G_IS_DBUS_CONNECTION (connection)' failed
[2016-07-26 23:14:37.468] [ERROR] EthereumNode - Failed to start node { Error: spawn /home/islab/mist/nodes/geth/linux-x64/geth ENOENT
    at exports._errnoException (util.js:949:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:182:32)
    at onErrorNT (internal/child_process.js:348:16)
    at _combinedTickCallback (internal/process/next_tick.js:74:11)
    at process._tickCallback (internal/process/next_tick.js:98:9)
  code: 'ENOENT',
  errno: 'ENOENT',
  syscall: 'spawn /home/islab/mist/nodes/geth/linux-x64/geth',
  path: '/home/islab/mist/nodes/geth/linux-x64/geth',
  spawnargs: [ '--fast', '--cache', '512', '--support-dao-fork' ] }
[2016-07-26 23:14:37.482] [ERROR] main - Error starting up node and/or syncing { Error: spawn /home/islab/mist/nodes/geth/linux-x64/geth ENOENT
    at exports._errnoException (util.js:949:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:182:32)
    at onErrorNT (internal/child_process.js:348:16)
    at _combinedTickCallback (internal/process/next_tick.js:74:11)
    at process._tickCallback (internal/process/next_tick.js:98:9)
  code: 'ENOENT',
  errno: 'ENOENT',
  syscall: 'spawn /home/islab/mist/nodes/geth/linux-x64/geth',
  path: '/home/islab/mist/nodes/geth/linux-x64/geth',
  spawnargs: [ '--fast', '--cache', '512', '--support-dao-fork' ] }

```

I did install Mist by  ethereum/mist/README.md.
But I saw 'couln't connect to node' message for one day..

I've checked the existing issues.
So I tried to check things, I don't solve this problem.

**Comment by luclu** [comment]
It looks like you didn't pull the nodes into your repo.
Please try to run:
```bash
gulp update-nodes
```

**Comment by jiiiiiiiin** [comment]
@luclu Thanks for answering. I did what you told me to do.

---------------------------------------------
root@ubuntu:/home/islab/mist/interface# gulp update-nodes
[23:56:03] Working directory changed to /home/islab/mist
You can select a platform like: --platform (all or darwin or win32 or linux)
Mist version: 0.8.1
Electron version: 1.2.5
[23:56:04] Using gulpfile /home/islab/mist/gulpfile.js
[23:56:04] Starting 'clean:nodes'...
[23:56:04] Finished 'clean:nodes' after 6.59 ms
[23:56:04] Starting 'downloadNodes'...
root@ubuntu:/home/islab/mist/interface# 
---------------------------------------------

But the situation just doesn't change...

**Comment by luclu** [comment]
Hmm, it seems to not download the nodes.
Could you kindly try to run:
```
gulp update-nodes --platform linux
```

**Comment by luclu** [comment]
Fixed with https://github.com/ethereum/mist/pull/1046

**Comment by jiiiiiiiin** [comment]
@luclu I fixed and I executed the command..
 
,,,,,,,,javascript 
root@ubuntu:/home/islab/mist# gulp update-nodes
You can select a platform like: --platform (all or darwin or win32 or linux)
Mist version: 0.8.1
Electron version: 1.2.5
[04:12:12] Using gulpfile /home/islab/mist/gulpfile.js
[04:12:12] Starting 'clean:nodes'...
[04:12:12] Finished 'clean:nodes' after 18 ms
[04:12:12] Starting 'downloadNodes'...
[04:12:12] Downloading https://github.com/ethereum/go-ethereum/releases/download/v1.4.10/geth-OSX-20160716155225-1.4.10-5f55d95.zip...
[04:12:12] Downloading https://github.com/ethereum/go-ethereum/releases/download/v1.4.10/geth-Linux64-20160716160600-1.4.10-5f55d95.tar.bz2...
[04:12:12] Downloading https://github.com/ethereum/go-ethereum/releases/download/v1.4.10/Geth-Win64-20160716155900-1.4.10-5f55d95.zip...
[04:12:12] Downloading https://bintray.com/karalabe/ethereum/download_file?file_path=geth-1.4.10-stable-5f55d95-linux-386.tar.bz2...
[04:12:12] Downloading https://bintray.com/karalabe/ethereum/download_file?file_path=geth-1.4.10-stable-5f55d95-windows-4.0-386.exe.zip...
[04:12:20] Downloaded https://bintray.com/karalabe/ethereum/download_file?file_path=geth-1.4.10-stable-5f55d95-linux-386.tar.bz2 after 7.56 s
[04:12:20] Downloaded https://bintray.com/karalabe/ethereum/download_file?file_path=geth-1.4.10-stable-5f55d95-windows-4.0-386.exe.zip after 8.31 s
[04:12:28] Downloaded https://github.com/ethereum/go-ethereum/releases/download/v1.4.10/geth-OSX-20160716155225-1.4.10-5f55d95.zip after 16 s
[04:12:41] Downloaded https://github.com/ethereum/go-ethereum/releases/download/v1.4.10/geth-Linux64-20160716160600-1.4.10-5f55d95.tar.bz2 after 29 s
[04:12:42] Downloaded https://github.com/ethereum/go-ethereum/releases/download/v1.4.10/Geth-Win64-20160716155900-1.4.10-5f55d95.zip after 30 s
[04:12:42] Finished 'downloadNodes' after 30 s
[04:12:42] Starting 'unzipNodes'...
[04:12:46] Finished 'unzipNodes' after 4.1 s
[04:12:46] Starting 'renameNodes'...
/geth-1.4.10-stable-5f55d95-linux-386
/geth-1.4.10-stable-5f55d95-windows-4.0-386.exe
[04:12:46] Finished 'renameNodes' after 165 ms
[04:12:46] Starting 'renameNodesDeleteOld'...
[04:12:46] Finished 'renameNodesDeleteOld' after 38 ms
[04:12:46] Starting 'update-nodes'...
[04:12:46] Finished 'update-nodes' after 22 μs
root@ubuntu:/home/islab/mist#
,,,,,,,,,,,,,,

'node update' is improved before, but the issue doesn't solve...

**Comment by luclu** [comment]
Try now, this PR should solve your problem  
> Fix node download logic #1048

**Comment by jiiiiiiiin** [comment]
@luclu Thank you!!!!!!!!!!!!! It works!!!!
